diff --git a/PC/launcher.c b/PC/launcher.c
index 09ac7d902c..f97648cbf7 100644
--- a/PC/launcher.c
+++ b/PC/launcher.c
@@ -1,3 +1,9 @@
+#if defined __GNUC__ && __GNUC__ >= 14
+#pragma GCC diagnostic warning "-Wimplicit-function-declaration"
+#pragma GCC diagnostic warning "-Wincompatible-pointer-types"
+#pragma GCC diagnostic warning "-Wint-conversion"
+#pragma GCC diagnostic warning "-Wreturn-mismatch"
+#endif
 /*
  * Copyright (C) 2011-2013 Vinay Sajip.
  * Licensed to PSF under a contributor agreement.
